packages/kbn-es-types/src/search.ts

```diff
@@ -593,8 +593,7 @@ export type AggregateOf<
 
 export type AggregateOfMap<TAggregationMap extends AggregationMap | undefined, TDocument> = {
   [TAggregationName in keyof TAggregationMap]: Required<TAggregationMap>[TAggregationName] extends AggregationsAggregationContainer
-    ? // @ts-expect-error not sure how to fix this, anything I've tried causes errors upstream - Dario
-      AggregateOf<TAggregationMap[TAggregationName], TDocument>
+    ? AggregateOf<Required<TAggregationMap>[TAggregationName], TDocument>
```

Comment:
Does this not actually make things required where they should be optional?

Reply:
Talked on Slack. I'll see if I can maintain the optionality of this property

Reply:
@dgieselaar in [`8f812e8` (#179758)](https://github.com/elastic/kibana/pull/179758/commits/8f812e8f8c1711f65c0c0c83eb6d91f7c238da1d) I added some tests to confirm the expected behavior (despite using `Required` 🤷)

Reply:
Thanks, I went over it and I think it's correct!